lib/config.dart

```diff
@@ -1,24 +1,196 @@
+  // Write to .env file
+  var file = File(dotEnvPath);
+  var content = file.readAsStringSync();
+
+  String newContent;
+  if (!isNewValue) {
+    newContent =
+        content.replaceAll(RegExp("$name=.*", multiLine: true), "$name=$value");
+  } else {
+    newContent = "$content\n$name=$value";
+  }
+
+  file.writeAsStringSync(newContent);
+}
+
+String acceptStdIn(String key) {
+  print("Please enter a value for $key:");
+  while (true) {
+    var value = stdin.readLineSync();
+    if (value != null && value.isNotEmpty) {
+      return value;
+    }
+    print("Error: Please enter a value for $key:");
   }
+}
 
-  if (secret.length < 32) {
-    throw Exception('.env: JWT_AUTH_SECRET must be at least 32 characters long');
+bool getConfirmation(String message) {
+  print("$message (y/n)");
+  while (true) {
+    var value = stdin.readLineSync();
+    if (["y", "Y"].contains(value)) {
+      return true;
+    } else if (["n", "N"].contains(value)) {
+      return false;
+    }
+    print("Error: Please enter y or n");
   }
-  return secret;
 }
 
-String _rubixQuorumKeyPass() {
-  var pass = _env['RUBIX_QUORUM_KEY_PASS'];
-  if (pass == null) {
-    throw Exception('.env: RUBIX_QUORUM_KEY_PASS not set');
+class ConfigItem {
+  final String envVar;
+  late String value;
+  final String defaultValue;
+  final List<Validator> validators;
+  final bool optional;
+
+  ConfigItem(
+      {required this.envVar,
+      this.defaultValue = "",
+      this.validators = const [],
+      this.optional = false}) {
+    if (optional && defaultValue.isNotEmpty) {
+      throw Exception("Optional config items cannot have default values");
+    }
+
+    if (defaultValue.isNotEmpty) {
+      // Validate default values - This is to ensure that the default values we hardcode are valid. (Ideally, test case should be added)
+      for (var validator in validators) {
+        validator.throwExceptionIfInvalid(defaultValue);
+      }
+    }
+
+    if (_env.isDefined(envVar)) {
+      value = _env[envVar]!;
+    } else {
+      if (optional) {
+        bool answer = getConfirmation(
+            "Env: $envVar is optional, Do you want to set a value?");
+        if (answer) {
+          var newValue = acceptStdIn(envVar);
+          value =
+              validateAndAcceptInput(envVar, newValue, validators, fix: true);
+        } else {
+          value = "";
+        }
+      } else {
+        if (defaultValue.isNotEmpty) {
+          var confirmDefault = getConfirmation(
+              "Do you want to set the default value ($defaultValue) for $envVar?");
+
+          if (confirmDefault) {
+            value = defaultValue;
+          } else {
+            value = acceptStdIn(envVar);
+          }
+        } else {
+          value = acceptStdIn(envVar);
+        }
+
+        value = validateAndAcceptInput(envVar, value, validators, fix: true);
+      }
+    } // validators.forEach((validator) => validator.throwIfInvalid());
+    print("ConfigItem: $envVar = $value");
```

Comment:
Please remove the unwanted comment statements.

Reply:
Removed! @Allen-Cherian . Thanks for pointing it out